### DISCUSSION TITLE
Custom functions giving error

### DISCUSSION THREAD

**Discussion by pankhiprasher** [stoplightio/spectral#1897] [Q&A]
When running spectral cli command - >spectral lint --ruleset functions\.spectral.yaml  API_1.0.yaml  with custom functions inside functions folder and placed at the same folder level with the spectral yaml file , still throwing this error - **then.function is not a function**

What is this error and how do we resolve this

**Comment by P0lip** [comment]
This means that one of the custom functions you use is not listed among the `functions` array in your ruleset.

**Reply by pankhiprasher** [reply]
@P0lip This is the way i have my functions array in the ruleset file , is there something else i should be doing , because this ruleset seems to work perfect and is linting when i am doing it in spectral web platform . 

formats:
  - oas3
extends:
  - 'spectral:oas'
functions:
  - checkType
  - checkHeaderDefaultValue
  - checkduplicate
  - checkAcceptLanguage
  - unique-resourceType-name
  - checkErrorMessage

**Reply by P0lip** [reply]
To be able to tell whether this is valid or not, I'd need to see the contents of `rules` as well.
If you could send that as well, that'd be great.
If you don't want to share it here, feel free to send me an email -> jakub(at)stoplight.io

**Reply by pankhiprasher** [reply]
@P0lip Sure I'll send over the ruleset file to you
